src/H5FDpublic.h

```diff
@@ -249,9 +249,23 @@
  * the VFD stack en-route to the terminal VFD.
  * If that VFD does not support the ctl call, the
  * pass through VFD should fail or succeed as directed
- * by the  H5FD_CTL__FAIL_IF_UNKNOWN_FLAG.
+ * by the  H5FD_CTL_FAIL_IF_UNKNOWN_FLAG.
  */
-#define H5FD_CTL__ROUTE_TO_TERMINAL_VFD_FLAG 0x0002
+#define H5FD_CTL_ROUTE_TO_TERMINAL_VFD_FLAG 0x0002
+
+/* Backwards compatibility. Support old spelling with double underscore. */
+#if 1 /* #ifndef H5_NO_DEPRECATED_SYMBOLS */
+#define H5FD_CTL__INVALID_OPCODE              H5FD_CTL_INVALID_OPCODE
+#define H5FD_CTL__TEST_OPCODE                 H5FD_CTL_TEST_OPCODE
+#define H5FD_CTL__GET_MPI_COMMUNICATOR_OPCODE H5FD_CTL_GET_MPI_COMMUNICATOR_OPCODE
+#define H5FD_CTL__GET_MPI_RANK_OPCODE         H5FD_CTL_GET_MPI_RANK_OPCODE
+#define H5FD_CTL__GET_MPI_SIZE_OPCODE         H5FD_CTL_GET_MPI_SIZE_OPCODE
+#define H5FD_CTL__MEM_ALLOC                   H5FD_CTL_MEM_ALLOC
+#define H5FD_CTL__MEM_FREE                    H5FD_CTL_MEM_FREE
+#define H5FD_CTL__MEM_COPY                    H5FD_CTL_MEM_COPY
+#define H5FD_CTL__FAIL_IF_UNKNOWN_FLAG        H5FD_CTL_FAIL_IF_UNKNOWN_FLAG
+#define H5FD_CTL__ROUTE_TO_TERMINAL_VFD_FLAG  H5FD_CTL_ROUTE_TO_TERMINAL_VFD_FLAG
+#endif
```

Comment:
You can just do the rename and remove the double underscore compatibility versions you've defined here. HDF5 1.13.x is an experimental release and we don't promise compatibility between releases. The ctl callback is 1.13.x only, so we don't have to worry about 1.12, etc.

As Jordan commented earlier, we don't support compiling the library with a C++ compiler anymore, but this file will be consumed by anyone consuming the library from C++, so we do need to be mindful of C++ issues in the public headers.

Reply:
Done.